### PR TITLE
wgengine: stop conntrack log spam about Canonical net probes

### PR DESCRIPTION
**Pull request by bradfitz**
Like we do for the ones on iOS.

As a bonus, this removes a caller of tsaddr.IsTailscaleIP which we
want to revamp/remove soonish.

Updates #13687

### CODE REVIEW

File: wgengine/pendopen.go [hunk 1]
```diff
@@ -86,6 +86,35 @@ func (e *userspaceEngine) trackOpenPreFilterIn(pp *packet.Parsed, t *tstun.Wrapp
 	return
 }
 
+var (
+	appleIPRange     = netip.MustParsePrefix("17.0.0.0/8")
+	canonicalIPRange = netip.MustParsePrefix("185.125.190.0/24")
```
Comment (andrew-d) [inline wgengine/pendopen.go:91]: Should we be adding the other prefixes announced by [`AS41231`](https://bgp.he.net/AS41231)?

**Review by andrew-d — APPROVED** [review]

File: wgengine/pendopen.go [hunk 2]
```diff
@@ -86,6 +86,35 @@ func (e *userspaceEngine) trackOpenPreFilterIn(pp *packet.Parsed, t *tstun.Wrapp
...
+// spam like:
+func (e *userspaceEngine) isOSNetworkProbe(dst netip.AddrPort) bool {
+	// iOS had log spam like:
+	// open-conn-track: timeout opening (100.115.73.60:52501 => 17.125.252.5:443); no associated peer node
+	if runtime.GOOS == "ios" && dst.Port() == 443 && appleIPRange.Contains(dst.Addr()) {
+		if _, ok := e.PeerForIP(dst.Addr()); !ok {
+			return true
+		}
+	}
+	// NetworkManger; https://github.com/tailscale/tailscale/issues/13687
```
Comment (andrew-d) [inline wgengine/pendopen.go:108]: Oh, and this should probably be `NetworkManager` (not `NetworkManger`)

Comment (bradfitz) [reply wgengine/pendopen.go]: done

File: wgengine/pendopen.go (hunk 1 above)
Comment (bradfitz) [reply wgengine/pendopen.go]: Done. Added all. And a test.

**Pull request merged**